### DISCUSSION TITLE
Welcome to GliderTools Discussions!

### DISCUSSION THREAD

**Discussion by lukegre** [GliderToolsCommunity/GliderTools#47] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by soerenthomsen** [comment]
Hi everyone, 

Great idea to introduce ourselves here. I first want to thank everyone who contributed to GliderTools so far and set the ground! 

I am [Soeren Thomsen](https://www.researchgate.net), Physical Oceanographer. I have been working with Slocum gliders since around 2011, including the development of turbulence processing software for MicroRider measurements during my Bachelor at GEOMAR. For my PhD/PostDoc at GEOMAR I designed and carried out two glider fleet experiments off Peru (https://agupubs.onlinelibrary.wiley.com/doi/full/10.1002/2015JC010878, https://agupubs.onlinelibrary.wiley.com/doi/full/10.1002/2016GL070548) and Mauritania (https://bg.copernicus.org/articles/16/979/2019/). 

Big fan of open science. So all related datasets are quality controlled open access available on Pangaea (https://doi.pangaea.de/10.1594/PANGAEA.860727, https://doi.pangaea.de/10.1594/PANGAEA.860480) thanks to G. Krahmann and the data management team at GEOMAR 🙏. Please get in contact if you are interested in glider datasets in upwelling systems (i.e. off Peru, Mauritania, Namibia or Benguela). I can navigate you through the different available deployments available on Pangaea.

Right now I work 50 % as OceanGliders.org Best Practice coordinator. I am willing to contribute to GliderTools i.e.to ensure that OceanGliders.org Best Practices i.e. on Delayed Mode Quality Control (DMQC) are implemented into GliderTools. 

I switched to Python in 2018 mainly to analyse ROMS/CROCO model output using xarray, dask etc! So far I have little experience in collective coding but motivated to learn on the fly!

Looking forward to working with all of you!

Best 
Soeren

**Comment by callumrollo** [comment]
Hi all,

I'm Callum Rollo, early career physical oceanographer and Python/linux tinkerer. I have a few years experience with Seagliders and want to drive more community development of open source tooling. I take the Argo community as a great inspiration, particularly packages like [argopy](https://github.com/euroargodev/argopy). I haven't contributed to GliderTools yet, but aim to get it on conda-forge to increase visibility and ease of use.

**Comment by isgiddy** [comment]
Hello!

Isabelle Giddy here. I am a Phd student physical/ bio-physical oceanographer. Love the philosophy and community in open-source code development. I use GliderTools frequently for my research and have been/ still am tentatively contributing :) - initially worked on the gridding module. Now working to fix errors and update some of the existing modules. My current focus is one of best practice in that I want to develop the code I use in my research into generic modules that facilitates asking future science questions with glider data and learn along the way.

**Comment by marcelduplessis** [comment]
Hi everyone! 

I'm Marcel du Plessis, a postdoc at the University of Gothenburg. I worked with Seaglider data for my Masters and PhD and assisted in the conceptualisation and testing of GliderTools. I am new to contributing to the open-source community but I'm eager to learn and help it grow where I can! I've been using Python for a few years now and am extremely grateful to the community for making such amazing tools which I use almost every day.

**Comment by jbusecke** [comment]
Howdy, folks. 

I am Julius, an Assistant Research Scientist at Lamont/Columbia University. I have worked with many different obs platforms (but not gliders yet) in my career but am at the moment more focussed on working with Earth System Models. 

I really don't like to make the distinction between "Observational Oceanographers" and "Modellers", and assertions that one of them is just inherently more technically inclined. I firmly believe that future science needs both and that increasingly complex and interdisciplinary topics require our work to be reproducible, open, and collaborative. 
I think that most science these days is done in the [code we all use and write each day](https://www.chronicle.com/article/the-scientific-paper-is-outdated/). I hope I can bring some of my experience in open source development and open science to this project to help people to learn about the inner workings of package maintenance/CI etc. 

First and foremost I hope this effort will foster a community effort that avoids duplication of work and encourage more usage of the amazing data gliders provide to everyone from the 'veterans' to newcomers and even folks like myself, who don't work with this data on a day-to-day basis.

**Reply by soerenthomsen** [reply]
Thanks for your contribution Juls! 🙏

**Reply by jbusecke** [reply]
![image](https://user-images.githubusercontent.com/14314623/113084269-ab0b1f00-91ab-11eb-81ab-a2e790d803e5.gif)

**Comment by tomhull** [comment]
Hi glidertools

I'm Tom, marine observations scientist at the Cefas and the University of East Anglia (UK).
I have 10+ years experience with operational oceanography with autonomous platforms.
I'm mostly a R developer, having written oceanographic packages for processing mooring, CTD, ferrybox and glider data.
I do have some python experience.

My science background is in biogeochemistry and I have a particular interest in dissolved oxygen and optodes. 
Most recently I've been responsible for the Seaglider and Slocum data from a 18 month North Sea monitoring program (https://altereco.ac.uk/).

I'm very happy to contribute to this effort.

**Reply by soerenthomsen** [reply]
Hi @tomhull ! Nice to e-meet you! Great to hear that you are also working on oxygen and optodes!

**Comment by bastienqueste** [comment]
Hi all,
I'm Bastien Queste, Asst. Prof. at the University of Gothenburg. I've been a glider user since 2008-2009 ish. Originally very involved with Seagliders, now also using SeaExplorer gliders. I was lead developer of the UEA Seaglider toolbox for some 10 or so years, before leaving for Sweden. Matlab/Python user, keen to correct mistakes of the past and embrace open software. Scientific expertise revolves around biophysical interactions and how ocean physical processes impact ventilation and remineralisation in ocean oxygen minimum zones.

**Comment by AnthonyBosse** [comment]
Hi all,
I am Anthony Bosse, Physical Oceanographer. I am working with gliders since 2012 (Master, then PhD) in France and then Norway (Postdocc @ UiB, Bergen). I am now back in France at Aix-Marseille Univ. with a strong implication in long-term obs (MOOSE project) including two endurance glider lines in the NW Med.
My scientific interests is centered around Ocean mixing spanning from understanding the impact of small scale physics (turbulence) to large-scale and low-frequency ocean variability.
I have written a few scripts (in Matlab) on eddy-center detection from depth-average currents, glider flight model to retrieve vertical velocities, streamwise coordinates for geostrophic velocity calculation and instability regimes deduced from Potential Vorticity... Discovered Python while teaching wit my students and keen on switching for research :)
So I have a strong motivation to contribute on glider DMQC, develop open-source tools and science around gliders which can facilitate the work of glider users in the future (including myself).
Happy to e-meet some of you and eager to discover what we'll achieve together :)

**Reply by castelao** [reply]
Hi @AnthonyBosse , you might enjoy pyRings (https://github.com/castelao/pyrings) a package to estimate the eddy-center. It considers the deformation of the eddy due to the translation during the survey. That can be an important factor since the gliders are relatively slow. It's old but should work fine.

**Comment by patricialg** [comment]
Hi all,
My name is Patricia and I work as Sensor Science Delivery Lead at NOC in Southampton. I'm currently working with nutrient sensors in the lab and as delivery lead for some projects like TechOceanS, AutoNuts, etc.. In these projects we develop sensors that are integrated in differnt platforms. In TechOceanS project, I'm also involved in the work related with the development of Best Practices for sensors.
I'm a marine scientist. I have extensive laboratory experience including some of the latest marine chemistry and trace analysis techniques. 
From 2014-2019 I was responsible for collection and analysis of samples from the European Station for Time series in the Ocean Canary Islands (ESTOC) and the Test Site of PLOCAN in addition to the calibration and validation of sensors integrated in moorings and gliders. I used to work closely with glider pilots (I was the lab rat in a white/mostly white lab coat).
I'm contributing (or trying to contribute) to the Glider best practices document that Soeren is coordinating. I'm helping with the Oxygen sensors section. My main interest is to help, when possible, to calibrate/validate sensors so that they will produce the best possible quality data and find the best protocol for the gold standard method to achieve this. Gliders are the future of Ocean Observation but without good sensors, they are only fancy toys, so I really believe that this initiative is very important.
I'm happy to discuss, learn from others and look together for the best approaches :)

**Comment by dhruvbalwada** [comment]
Hi Everyone, 

I am Dhruv Balwada, a physical oceanographer. I am currently a postdoctoral researcher at the University of Washington (not with the glider people here). My research generally focuses on ocean turbulence at the meso and submesoscales, and over the years I have really enjoyed describing "the shape of water" (essentially quantifying variability at different scales) among other things. 
 
I have only started working with glider data since late 2019, am almost completely new to this enterprise, and wanting and willing to learn. I transitioned to gliders after having spent some time working closely with the teams that were developing Pangeo, and so as soon as I started I looked for an open source toolbox. I was very glad to have found Glidertools, and am now even more excited by the rapid rise in engagement. I think these open source tools not only provide easy access to code and save huge amount of time, but also help build better frameworks to think in - essentially collecting together the best ideas from many people that rise out of collective trial and error. It also helps democratize the process, as I don't need to be working with the one group that "does things best" anymore, and experience can easily be shared and transferred. 

In terms of working with gliderdata, I am very interested in the mapping and visualization side of things. I feel that since gliders are starting to look at scales that have traditionally not been accessible to oceanographers, we need to reevaluate how we think and work with these spatio-temporally complex sampling patterns. I am also starting to wonder about what are the best ways to compare glider data with high resolution simulations, specifically thinking beyond the fact that the ocean models at high resolutions do well because the wiggles from the high resolution models are qualitatively starting to look similar to those from gliders. 

Looking forward to collaborating with folks here.

**Comment by sarahnicholson** [comment]
Hi!  

My name is Sarah Nicholson, i'm a Researcher at Southern Ocean Carbon-Climate Observatory (SOCCO) based at the CSIR, South Africa, Cape Town.  Iv been working with Seagliders, Slocums and Wave Gliders since 2017 and I currently co-Lead glider activities at SOCCO. Im particularly interested in microstructure from gliders, twinned glider experiments and understanding the coupled physics-carbon system. I contributed towards conceptual development of GliderTools and am an avid user. Happy to continue to contribute where I can.

**Comment by sebswart** [comment]
Hi! 
I'm Seb Swart, Assoc Prof at Uni Gothenburg, working with Seagliders, Wave Gliders, Sailbuoys, Saildrones, since 2011. Started with gliders in S Africa at SOCCO and now I am in Sweden where we have built up our fleet of Swedish-based gliders and ASVs (more news on Twitter @PolarGliders). I am advisor to Voice of the Ocean, which now deploys numerous gliders and ASVs in the Baltic Sea. My research areas using gliders are mostly around upper ocean physics, submesoscale processes and air sea fluxes, mostly focused on the Southern Ocean. Advocate :) and involved in GliderTools - I feel strongly this requires to be adopted and used by the community so we have common formats for processing glider data (post basestation), where it makes sense of course. Happy to meet you all!

**Comment by tjryankeogh** [comment]
Hi everyone,

My name is Tommy Ryan-Keogh, I am a postdoctoral research fellow at SOCCO-CSIR based in Cape Town, South Africa. I am an original contributor and code developer of GliderTools. I have developed other packages in python as a part of SCOR activities, and want to see GliderTools grow into a community tool that is constantly growing as our needs and requirements do. I have been working with Gliders since 2015 with a specific focus on bio-optics to examine biogeochemistry (phytoplankton production, carbon export).

**Comment by AlicePietri** [comment]
Hello everyone, 

my name is Alice Pietri, I'm a physical oceanographer working at IMARPE in Peru. I've been working with gliders since I started my phD a decade ago in France (LOCEAN) and then in Germany (GEOMAR). I mainly have experience with slocums, from preparation to deployment to data processing. So far I've only been using Matlab but I have been meaning to switch to Python for some time now and I hope I can get involved and contribute. I'm very motivated to build a more open environment to facilitate glider data processing.  
Looking forward to interacting with you here.

**Reply by castelao** [reply]
Hi @AlicePietri , I was on that glider workshop at IMARPE a few years ago organized by Dimitri. The Carpentries have a nice introduction to Python (https://software-carpentry.org/lessons/)

**Comment by lukegre** [comment]
Hi all, 

I'm [Luke Gregor](https://www.researchgate.net/profile/Luke-Gregor). I am based at ETH Zürich doing a postdoc with a focus on global carbon fluxes and ocean acidification. 
Unlike most of the group here I do not work with gliders, but I, along with the OGs (Tommy, Marcel, Sarah, Izzy, Seb), am one of the original developers of GliderTools. Since moving to ETH, my time for GliderTools has dwindled to near nothing 😢 but I would still like to be involved! So, I'll be lurking around the discussions and doing a bit of coding and reviewing.

**Comment by dennisstanley64** [comment]
I'm Dennis Stanley I'm based at University of Western Australia and i'm the Operations manager for [IMOS](https://imos.org.au/facilities/oceangliders) ocean glider program.  We have completed hundreds of glider missions and created documented processing protocols all publicly available through the website and Australian Ocean Data Network (AODN). I'm an electrical engineer by training and have a tinkerers level knowledge of coding in python having processed ocean data from gliders and ocean models and moorings and ...... on my pet subject which is the physical oceanographic drivers of coral bleaching and in particular the role which ocean glider data, ocean models , and insitu logging can play in revealing ; the vertical structure of the ocean, turbulence and benthic flow in relation to coral bleaching. Huge repositories of data from various sources exist on servers around the world and i believe there is a technical hurdle of fair degree of computer & data science literacy required to access and make sense of them . This is leading to our understanding of ocean processes being skewed by the accessibility of data. I'd love to see the barriers to access of this data lowered by readily available documented tools which could allow a broader insight into the processes driving change in our oceans.

**Reply by soerenthomsen** [reply]
Welcome Dennis Stanley (@dennisstanley64)! Great to have you on board! Your work at IMOS is impressive!

**Reply by nmalan** [reply]
@dennisstanley64, have you experimented at all with getting the IMOS glider netcdfs to play nice with GliderTools? I've been messing around with this (with limited success, some things work, others don't), and was wondering if anyone else had trodden this path before?

**Comment by svenja-ryan** [comment]
Hi everyone,

this is a great initiative!

I am Svenja and a postdoc at Woods Hole Oceanographic Institution. I have not worked with gliders before but will likely start using/processing glider data sometime this year. After using Matlab for almost 10 years, I switched to Python when I started my postdoc in 2019 where I am working with large model outputs - have not regretted it!! The glider person at WHOI uses Matlab, so I am definitely keen to see where GliderTools is going and compare tools. Hopefully of course I hope to be able to contribute along the way. I got introduced to collective coding during OceanHackWeek, which was a lot of fun.

Currently I am studying marine heatwaves, their depth structure and regional drivers and recently shifted my focus on the Northeast US continental shelf, which is the region where I will be analysing glider data.

Looking forward to the interactions and discussion here and hope I can be of help!

**Reply by dennisstanley64** [reply]
Hi Svenja. marine heatwaves. and in particular their depth is a great current topic (in Australia anyway) and gliders can contribute in a fantastic way. many SST heatwaves are really, thin mixed layer events. The water column temp is, depth average,  unchanged. Others are advection events or .... ocean glider data (and argo floats & modelling) can reveal the characteristics of a marine heatwave. we have a marine heatwave event based glider program and we see advection events; like the strengthening of the east Australian current bringing warmer water to the Tasmanian coast  https://anfog.ecm.uwa.edu.au/slocum_portal/unit287/TasEastCoast-09Mar21/unit287_TasEastCoast-09Mar21.kmz
others are more insitu low wind events. python is a great tool to code in because you can contribute globally without essentially locking  your intellectual property to a platform for others profit from which is what i think matlab does.

**Reply by soerenthomsen** [reply]
Hi Svenja (@sryan288), 

Great to have you in! Which glider platforms are used at WHOI? You research topic sounds exciting. Please check out https://ebuswebinars.wixsite.com/ebuswebinars where we have talks about upwelling systems every months. There are also old webinars to watch. Best Soeren

**Reply by svenja-ryan** [reply]
Hi Soeren (@soerenthomsen),

thanks - will check out the seminars! As far as I know it is mainly Spray gliders that are used, which were developed at WHOI and Scripps (http://gliders.whoi.edu/instrumentation/spray/). Not having worked with gliders, I am not sure how the differ from e.g. Slocum - should probably read up on that once I start working with the data ;-)

**Reply by castelao** [reply]
Hi @sryan288 ,
do you work with Robert Todd? This is Gui from Scripps. I was at the OceanHackWeek as well. I'm the one who developed the Python infrastructure for Spray. Let me know how I can help you. Spray are the orange ones!

**Comment by LewisDrysdale** [comment]
Hi GliderToolsCommunity,

I am Lewis and I work as a robotics support scientist at the Scottish Marine Robotics Facility based at the Scottish Association for Marine Science. We work almost exclusively with Seagliders deployed around Scotland, the North Atlantic, and Arctic Ocean. Our data processing is mostly done 'in house' by Postdocs on various projects but we are now looking to adopt a toolbox for all processing and data delivery. GliderTools could be perfect for us. No-one in our team uses Python, but now may a good time to start. I hope I will be able to contribute once I have learned how to use the toolbox. For now, is there an IDE folks can recommend for beginners using this GliderTools?

**Reply by callumrollo** [reply]
Welcome to the party Lewis! Good to have some SAMS representation. Personally I'm a big fan of [Pycharm](https://www.jetbrains.com/pycharm/) but it can be pretty overwhelming if you're just starting out with Python. I found [spyder](https://www.spyder-ide.org/) very useful when transitioning from matlab

**Reply by LewisDrysdale** [reply]
Thanks! Going to try with Pycharm, and see how I get on,

**Reply by jbusecke** [reply]
Not quite as powerful (yet) as most IDEs but I can recommend checking out [JupyterLab](https://jupyterlab.readthedocs.io/en/stable/getting_started/overview.html). This is used very widely on cloud computing instances like the [pangeo one](https://pangeo.io/cloud.html), but also on JupyterHub Deployments on various HPC centers, and you can run it locally. It makes it pretty neat to switch between systems, all you need is your webbrowser!

**Reply by bastienqueste** [reply]
+1 for JupyterLab. I used Matlab's IDE before, then used PyCharm a lot, but have now switched to JupyterLab entirely for this kind of analysis. Much more interactive, and somewhat more intuitive for ex-Matlabers.

B

**Reply by dhruvbalwada** [reply]
+1 for JupyterLab too. That is what I used when I transitioned over from MATLAB.

**Reply by LewisDrysdale** [reply]
Thanks all. I'm finding Pycharm a bit cluttered and confusing. Jupyter, on the other hand, is really great!

**Comment by backeb** [comment]
Hi all,
My name is Bjorn Backeberg, I am a researcher at Deltares' Software Centre in the Netherlands. I am a physical oceanographer interested in research underpinning operational oceanography. I have worked with a wide range of data and models, including data coming autonomous platforms. 
Over the last couple of years I have been working with cloud computing infrastructures around leveraging cloud technologies to scale our scientific results. Would love to support your activities, help with developments where possible and help grow the community! I love the community approach y'all have taken here! Excellent initiative!

**Comment by nlbeaird** [comment]
Hi all, 
My name is Nick Beaird, I'm a physical oceanographer at Rutgers University. I used gliders (seagliders) extensively for my PhD work at the University of Washington, but have been off in other observational worlds for a few years now. I'm starting to get back into gliders again and am really excited to see this effort!  I switched to using python a few years ago, and while I have benefited enormously from the  all the excellent open source work people are doing, I don't have experience contributing myself. However I'm excited to try to help and to follow along with this group!

**Reply by soerenthomsen** [reply]
Hi @nlbeaird! Great you are joining the effort. Have you worked on specific variables ?

**Reply by nlbeaird** [reply]
Hi @soerenthomsen, thanks! 

 I've mainly worked with standard (unpumped) CTD data, mostly to look at overflows ( eg: https://doi.org/10.1175/JPO-D-13-029.1 , https://doi.org/10.1175/JPO-D-12-094.1, https://doi.org/10.1002/2015JC011501) . Part of that was early work trying to infer dissipation of TKE from glider vertical velocity - something I'm still interested in. 

Recently I've been working with a student on some simple processing for active acoustics on gliders to look at zooplankon distributions (some messing around here: https://github.com/nlbeaird/glider_acoustics)

**Reply by soerenthomsen** [reply]
Great work @nlbeaird ! I am happy we connected here. I will get into contact with you also related to some OceanGliders.org Best Practice efforts!

**Comment by bastienqueste** [comment]
Nice to see the community growing! Welcome everyone :)

**Comment by castelao** [comment]
Hi all,

This is Gui Castelão from Scripps. I'm a physical oceanographer, and I do part of the Spray data pipeline, from satellite communications (https://github.com/castelao/iridiumSBD) to the GDAC & public data files (https://spraydata.ucsd.edu). I developed the Python infrastructure for Spray, also using xarray. If there is interest, I would be happy to merge or at least work to keep cross-compatibility. I also developed a package for automatic QC of oceanographic data, including gliders, that could be easily connected here (https://github.com/castelao/CoTeDe). I've been involved with Open Source for a while, so I'm excited to see this movement here.

**Reply by soerenthomsen** [reply]
Great to have you on board Gui (@castelao)! This is motivating! 

I am sure there is interest to merge where it makes sense and/or keep cross-compatibility. I am happy to facilitate this discussion and started an overview figure for a new package structure. This could be a good starting point to discuss together what could be merged and which tools could be made compatible. I will start a new discussion threat soonish on this.

**Comment by smerckel** [comment]
Hi,

I am Lucas Merckelbach, an oceanographer at Helmholtz Center Hereon, Geesthacht, Germany. (formally know as HZG). Here I work in a small group mostly focused on processes in stratified shelf seas, in particular turbulence, in relation to the effects of offshore wind farms. Started to work with Slocum gliders back in 2006, then at NOC in Southampton, UK, which was also the time I ditched matlab in favour of python. I am the author of dbdreader (https://github.com/smerckel/dbdreader/), which is a python module to read the binary files generated by Slocum gliders. As I understand it, GliderTools lacks a proper support for Slocum glider data, so I am happy to put in my expertise to improve the interface to Slocum data. 

I also wrote the python module gliderflight (https://github.com/smerckel/gliderflight), which can be used to compute glider speed through water, as well as vertical water velocities. The speed through water is an important input parameter in the processing of microstructure data obained from gliderborne microstructure sensors. It may be of interest to some of you.

**Reply by soerenthomsen** [reply]
Hi @smerckel, 

Great to have you onboard! Indeed it would be very important for GliderTools to be able to import Slocum glider data. I tried to make a first proposal for a new package structure here: #107. I assume these activities would also need to aligne with the new OceanGliders format mentioned here #49 and #50.

**Reply by jklymak** [reply]
Note [pyglider](https://github.com/c-proof/pyglider) is now using dbdreader (optionally).  dbdreader is much more robust and fast than the tools we were using before to do the raw conversion.

**Comment by LouClement** [comment]
Hi all,

I’m Louis Clement, I am a physical oceanographer and I work at the NOC in Southampton, UK. 
I’ve been working with Gliders (Seagliders) over the last year to investigate deep convection in the Labrador Sea. 
I used GliderTools for the QC and I was pleased to see the effort that was made in building the tools.
It’s great to see this momentum in trying to have uniform tools for the different platforms as it looks like many groups are reproducing the same work. I’d also be interested in joining the discussions.

**Reply by jbusecke** [reply]
I know that dude! Great to have you here, Louis!

**Reply by LouClement** [reply]
Thank you Julius, great to see you around as well.

**Comment by fabien-roquet** [comment]
Hello everyone,
Fabien Roquet, Prof. at Gothenburg University. I have worked for a long time with elephant seal tags in the Southern Ocean. I am the maintainer of MEOP (meop.net). Mostly with matlab originally. But now I am convinced that python is the future, and GliderTools look really promising to analyse hydrographic profiles. And seals are far better gliders than any of your gliders ^^, so GliderTools might contribute to animal-borne ocean sensors as well!

**Reply by soerenthomsen** [reply]
Welcome @fabien-roquet ! Indeed there must be a lot of similarities in how to treat the data from seals and gliders. For sure seals have even longer lasting battery packages! ;)

**Comment by charlotte-aj-williams** [comment]
Hi everyone,

My name is Charlotte Williams, I am a Research Scientist in the Marine Physics and Climate Group at the National Oceanography Centre, working primarily with our fleet of Gliders (since 2015). I have been involved in >40 glider deployments as part of the shelf sea biogeochemistry project (https://www.uk-ssb.org/), as well as being responsible for the slocum glider data for the AlterECO project (https://altereco.ac.uk/).  I have a multidisciplinary background and am interested in understanding biophysical interactions. I am particularly interested in using gliders to investigate oxygen depletion and dynamics in the coastal and shelf sea, and how this might respond to a changing climate. I have experience in using Aanderaa oxygen optodes that have been deployed on slocum gliders and am interested in finding the best practice to get the best possible oxygen data from gliders. I am also a co-WP lead on the EU GROOM II (gliders for research, ocean observation and management) project (https://www.groom-h2020.eu/).

**Comment by kimmotikka** [comment]
Hi everyone, 
I am Kimmo Tikka. I have operated Finnish Meteorological Institute's Slocum G2 glider in the Baltic Sea since 2016. I've participated GROOM-I and II and also marine data projects SeaDataNet and EMODNET Chem & Ingestion. 
I've used several matlab and python codes (SOCIB, Rutgers, ...) to handle glider data. Nowadays, I'm using mostly code of my own, just to know how the data is actually manipulated. Problems I have encountered are normal optode time lags and fluorometer data issues (filtering spikes, interpolating). I have also some tools of my own to plan and monitor the mission.

**Reply by kimmotikka** [reply]
The tools I use mostly are python and jupyter notebook with pandas and sqlalchemy & sqlite db. I think Julia language would  also be worth to notice.

**Comment by dennisstanley64** [comment]
i haven't tried glider tools (insert embarrassed emoji ) we have our own publicly available tool called glidescope which can be use to browse data and plot things up. generally i use this to roughly explore data then load it into python for more detailed analysis.
If the problem is opening the netcdfs in python. then there is a work around for this and that is to load them in via xarray.
here is some code i use to import three net cdfs into pandas data frames.the key bit is the last lines where i pass the netcdf data into xarray and then doing a to_dataframe from xarray instead of from netCDF directly. i don't know why this is more error tolerant but the direct import from netCDF doesn't work but via xarray does. 



import pandas as pdimport xarray as xrfrom netCDF4 import Dataset

# now get glider data gliderdatafilenameDec="IMOS_ANFOG_BCEOPSTUV_20191113T012730Z_SL287_FV01_timeseries_END-20191205T005735Z.nc"gliderdatafilenameJan="IMOS_ANFOG_BCEOPSTUV_20200123T021637Z_SL287_FV01_timeseries_END-20200224T233110Z.nc"gliderdatafilenameFeb="IMOS_ANFOG_BCEOPSTUV_20200227T032645Z_SL287_FV01_timeseries_END-20200327T232039Z.nc"# open the dataset
columns_to_drop =['PLATFORM', 'DEPLOYMENT', 'SENSOR1', 'SENSOR2', 'SENSOR3', 'SENSOR4',       'TIME_quality_control', 'LATITUDE', 'LATITUDE_quality_control',       'LONGITUDE', 'LONGITUDE_quality_control', 'HEAD',       'HEAD_quality_control', 'UCUR', 'UCUR_quality_control', 'VCUR',       'VCUR_quality_control', 'UCUR_GPS', 'UCUR_GPS_quality_control',       'VCUR_GPS', 'VCUR_GPS_quality_control', 'PHASE',       'PHASE_quality_control', 'PROFILE', 'PROFILE_quality_control', 'PRES',       'PRES_quality_control', 'DEPTH_quality_control',        'TEMP_quality_control', 'CNDC', 'CNDC_quality_control',        'PSAL_quality_control', 'DOX2', 'DOX2_quality_control', 'DOX1',       'DOX1_quality_control', 'CPHL', 'CPHL_quality_control', 'CDOM',       'CDOM_quality_control', 'VBSC', 'VBSC_quality_control', 'BBP',       'BBP_quality_control', 'IRRAD443', 'IRRAD443_quality_control',       'IRRAD490', 'IRRAD490_quality_control', 'IRRAD555',       'IRRAD555_quality_control', 'IRRAD670', 'IRRAD670_quality_control']  # columns to leave out of the dataframe to stop gobbling memory
gliderdataDec= xr.open_dataset(gliderdatafilenameDec, drop_variables = columns_to_drop).to_dataframe() #load netcdf into pandas dataframe via xarray#gliderdata.append(xr.open_dataset(gliderdatafilenameJan, drop_variables = columns_to_drop).to_dataframe())gliderdataJan= xr.open_dataset(gliderdatafilenameJan, drop_variables = columns_to_drop).to_dataframe() #load netcdf into pandas dataframe via xarraygliderdataFeb= xr.open_dataset(gliderdatafilenameFeb, drop_variables = columns_to_drop).to_dataframe() #load netcdf into pandas dataframe via xarray

 Hope this helpsDennis

    On Wednesday, 10 November 2021, 01:15:58 pm AEST, Neil Malan ***@***.***> wrote:  
 
 


@dennisstanley64, have you experimented at all with getting the IMOS glider netcdfs to play nice with GliderTools? I've been messing around with this (with limited success, some things work, others don't), and was wondering if anyone else had trodden this path before?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by nmalan** [reply]
Thanks Dennis, our current glider workflow is similar with xarray and pandas, we have about half of glider tools functionality working (everything in the vertical is fine, horizontal not so much). We're hoping to write a reader for the IMOS format, will share if we get it right!

**Reply by soerenthomsen** [reply]
Dear @dennisstanley64 and @nmalan thanks for your input. We try to keep our discussions as organised as possible. This is the discussion thread. I transferred your discussion to a new topic. Please adapt accordingly (i.e. title etc). And remove the thread here. 

Thanks a lot.

**Comment by dennisstanley64** [comment]
hmm by horizontally you mean XY (east north) because mostly our data is indexed temporally. Which i think is more accurate . given we KNOW what time a reading was made but we don't KNOW where it was made, exactly. in the future others may model our flight date better than us and obtain a better approximation of position and we shouldn't pre empt that with our current arbitrary approximation of position. Time is what we know to be correct and so how measurements should be indexed IMHO . 
    On Wednesday, 10 November 2021, 02:17:15 pm AEST, Neil Malan ***@***.***> wrote:  
 
 


Thanks Dennis, our current glider workflow is similar with xarray and pandas, we have about half of glider tools functionality working (everything in the vertical is fine, horizontal not so much). We're hoping to write a reader for the IMOS format, will share if we get it right!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by jklymak** [comment]
Hi, I'm Jody Klymak, Professor at University of Victoria, and principal PI for https://cproof.uvic.ca. Apparently I am also our lead developer, and in particular checkout https://github.com/c-proof/pyglider which purports to make CF-compliant NetCDFs from your glider data using a short python script and yaml metadata.

**Comment by MartinMohrmann** [comment]
Hi, I am Martin Mohrmann, Baltic Marine Researcher for the Voice of the Ocean Foundation. Originally from northern Germany, I did my PhD at the University of Gothenburg in physical oceanography and continue to work in Gothenburg for VOTO now. I have hardly worked with Gliders before, but I do it now and GliderTools is a great way to get started with the analysis. I also just wrote by first contribution. ;)

**Reply by soerenthomsen** [reply]
Great that you have joined Martin and thanks for supporting GliderTools!

**Comment by PrasannaPal21** [comment]
Hi, I'm Prasanna. I'm a CS student looking into the QARTOD QC project(ioos_qc) for GSoC 2026. I've been going through gliderTools to get a better understanding of how glider data is processed. Wanted to introduced myself here since the readme pointed me to this thread. I'm planning to contribute where I can and will probably start with improving the tests.

**Reply by MartinMohrmann** [reply]
Welcome Prasanna. Thank you for your first contribution to the project! 
Improving the tests is great! Our aim should be that GliderTools works well with most (all) gliderdata, for example from the [IOOS GDAC](https://gliders.ioos.us/erddap/index.html), [OceanGliders GDAC](https://www.seanoe.org/data/00453/56509/) or [VOTO](https://erddap.observations.voiceoftheocean.org/erddap/index.html).

**Reply by PrasannaPal21** [reply]
Hello @MartinMohrmann 
Can you please review the PR I made recently. #209.
